src/BTVNanoCommissioning/utils/correction.py

```diff
@@ -160,71 +271,128 @@ def load_pu(campaign, path):
 
 
-### Lepton SFs
-def eleSFs(ele, campaign, path):
-    _ele_path = f"BTVNanoCommissioning.data.LSF.{campaign}"
-    ext = extractor()
-    with contextlib.ExitStack() as stack:
-        real_paths = [
-            stack.enter_context(importlib.resources.path(_ele_path, f))
-            for f in path.values()
-        ]
-        ext.add_weight_sets(
-            [
-                f"{paths} {file}"
-                for paths, file in zip(path.keys(), real_paths)
-                if "ele" in paths
-            ]
+def puwei(correct_map, nPU):
+    if "correctionlib" in str(type(correct_map["PU"])):
+        return correct_map["PU"][list(correct_map["PU"].keys())[0]].evaluate(
+            nPU, "nominal"
         )
+    else:
+        return correct_map["PU"](nPU)
+
+
+def btagSFs(jet, correct_map, SFtype, syst="central"):
+    if "correctionlib" in str(type(correct_map["btag"])):
+        if SFtype.endswith("C"):
+            ## hard coded
+            if "DeepJetC":
+                return correct_map["ctag"]["deepJet_shape"].evaluate(
+                    syst, jet.hadronFlavour, jet.btagDeepFlavCvL, jet.btagDeepFlavCvB
+                )
+            else:
+                return correct_map["ctag"]["deepCSV_shape"].evaluate(
+                    syst, jet.hadronFlavour, jet.btagDeepCvL, jet.btagDeepCvB
+                )
+        else:
+            if "DeepJetB":
+                return correct_map["btag"]["deepJet_shape"].evaluate(
+                    syst, jet.hadronFlavour, abs(jet.eta), jet.pt, jet.btagDeepFlavB
+                )
+            if "DeepCSVB":
+                return correct_map["btag"]["deepJet_shape"].evaluate(
+                    syst, jet.hadronFlavour, abs(jet.eta), jet.pt, jet.btagDeepB
+                )
+    else:
+        if SFtype.endswith("C"):
+            if SFtype == "DeepJetC":
+                return getSF(
+                    jet.hadronFlavour,
+                    jet.btagDeepFlavCvL,
+                    jet.btagDeepFlavCvB,
+                    correct_map["ctag"][SFtype],
+                    syst,
+                )
+            else:
+                return getSF(
+                    jet.hadronFlavour,
+                    jet.btagDeepCvL,
+                    jet.btagDeepCvB,
+                    correct_map["ctag"][SFtype],
+                    syst,
+                )
 
-    ext.finalize()
-    evaluator = ext.make_evaluator()
+        else:
+            if SFtype == "DeepJetC":
+                return correct_map["btag"][SFtype].eval(
+                    syst, jet.hadronFlavour, abs(jet.eta), jet.pt, jet.btagDeepFlavB
+                )
+            else:
+                return correct_map["btag"][SFtype].eval(
+                    syst, jet.hadronFlavour, abs(jet.eta), jet.pt, jet.btagDeepB
+                )
+
+
+### Lepton SFs
+def eleSFs(ele, correct_map):
     ele_eta = ak.fill_none(ele.eta, 0.0)
-    ele_pt = ak.fill_none(ele.pt, 0.0)
+    ele_pt = ak.fill_none(ele.pt, 20.0)
```

Comment:
> Probably an easy answer, but why was this changed? 

This is due to the `correctionlib` cannot handle the events out of range, so change to minimum bin.

For the non-existing electron/muon, indeed this issue is not exist now(get rid of them in the following commit) since the current correction only extracted with the selected events. No arbitrary values filled now